src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -712,21 +712,6 @@ internal static ScriptBlock Create(Parser parser, string fileName, string fileCo
             return sbText;
         }
 
-        /// <summary>
-        /// Support for <see cref="ISerializable"/>.
-        /// </summary>
-        public virtual void GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
I presume the `ScriptBlock` class has `[Serializable]` attribute declared given that it has this virtual method. Please remove it as well.

Also, please search `[Serializable]` in the code base to make sure all are removed.

Reply:
There are many left, but we don't get a warning for those. The SYSLIB0051 specfically talks about classes deriving from System.Exception.

https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051#workaround

Reply:
Removed.